api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1260,8 +1277,17 @@ public ObjectStat statObject(String bucketName, String objectName)
            InternalException {
     HttpResponse response = executeHead(bucketName, objectName);
     ResponseHeader header = response.header();
-    return new ObjectStat(bucketName, objectName, header.lastModified(), header.contentLength(),
-                          header.etag(), header.contentType());
+    ObjectStat objectStat = null;
+
+    // Return object stat with key data if header has key and iv
+    if ((header.xamzMetaKey() != null) && (header.xamzMetaIv() != null)) {
```

Comment:
I had it like that, but @balamurugana has a valid comment about it breaking backwards compatibility - https://github.com/minio/minio-java/pull/542#discussion_r110333060